x-pack/plugins/serverless_search/public/application/components/connectors/connectors_table.tsx

```diff
@@ -0,0 +1,224 @@
+    {
+      text: i18n.translate('xpack.serverlessSearch.connectors.descriptionLabel', {
+        defaultMessage: 'Description',
+      }),
+      value: 'description',
+    },
+    { text: syncStatusLabel, value: 'last_sync_status' },
+  ];
+
+  const columns: Array<EuiBasicTableColumn<Connector>> = [
+    {
+      field: 'name',
+      name: nameLabel,
+      render: (name: string, connector: Connector) => (
+        <EuiLink href="TODO TODO TODO">{name || connector.id}</EuiLink>
+      ),
+      truncateText: true,
+    },
+    {
+      field: 'service_type',
+      name: typeLabel,
+      render: (serviceType: string | null) => {
+        const typeData = (connectorTypes?.connectors || []).find(
+          (connector) => connector.serviceType === (serviceType ?? '')
+        );
+        if (!typeData) {
+          return <></>;
+        }
+        return (
+          <EuiFlexGroup direction="row" gutterSize="s" alignItems="center">
+            <EuiFlexItem grow={false}>
+              <EuiIcon type={typeData.iconPath} aria-label={typeData.name} />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText size="s">{typeData.name}</EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        );
+      },
+    },
+    {
+      field: 'status',
+      name: connectedLabel,
+      render: (status: ConnectorStatus) =>
+        status === ConnectorStatus.CONNECTED ? (
+          <EuiIcon aria-label={connectedLabel} color="success" type="checkInCircleFilled" />
+        ) : (
+          <EuiBadge>
+            {i18n.translate('xpack.serverlessSearch.connectors.notConnectedLabel', {
+              defaultMessage: 'Not connected',
+            })}
+          </EuiBadge>
+        ),
+    },
+    {
+      field: 'status',
+      name: configuredLabel,
+      render: (status: ConnectorStatus) =>
+        [ConnectorStatus.CONNECTED, ConnectorStatus.CONFIGURED].includes(status) ? (
+          <EuiIcon aria-label={configuredLabel} color="success" type="checkInCircleFilled" />
+        ) : (
+          <EuiBadge>
+            {i18n.translate('xpack.serverlessSearch.connectors.notConfiguredLabel', {
+              defaultMessage: 'Not configured',
+            })}
+          </EuiBadge>
+        ),
+    },
+    {
+      field: 'last_sync_status',
+      name: syncStatusLabel,
+      render: (syncStatus: SyncStatus | null) =>
+        syncStatus ? (
+          <EuiBadge color={syncStatusToColor(syncStatus)}>{syncStatusToText(syncStatus)}</EuiBadge>
+        ) : (
+          <EuiBadge>
+            {i18n.translate('xpack.serverlessSearch.connectors.notSyncedLabel', {
+              defaultMessage: 'Not synced',
+            })}
+          </EuiBadge>
+        ),
+    },
+  ];
+
+  const items =
+    data?.connectors
+      .filter((connector) =>
+        filter ? `${connector[filter]}`.toLowerCase().includes(query.toLowerCase()) : true
+      )
+      .slice(pageIndex * pageSize, pageIndex + 1 * pageSize) ?? [];
```

Comment:
`pageIndex + 1 * pageSize` may fail due to the operator precedence.

3 + 1 * 10 = 13

Reply:
Fixed in latest